### PR TITLE
Add sidenav Accessibility improvements

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 88194935

## Summary

This PR improves the AX on the sidebar

## Dependencies

https://github.com/apple/swift-docc-render/pull/51

## Testing

Make sure you download and unzip the .doccarchive - [SlothCreator.doccarchive.zip](https://github.com/apple/swift-docc-render/files/8082216/SlothCreator.doccarchive.1.zip)

Steps:

1. run VUE_APP_DEV_SERVER_PROXY=/path/to/SlothCreator.doccarchive npm run serve
2. Visit http://localhost:8080/documentation/slothcreator
3. Try using Tabbing or VO

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: src/components/Navigator/NavigatorCard.vue [hunk 1]
```diff
@@ -472,6 +476,22 @@ export default {
         this.$refs.scroller.scrollToItem(index);
       }
     },
+    // TODO: This is very much a test
```
Comment (mportiz08) [inline src/components/Navigator/NavigatorCard.vue]: Is this comment still relevant?

File: src/components/Navigator/NavigatorCard.vue [hunk 2]
```diff
@@ -472,6 +476,22 @@ export default {
...
+    handleBlur(event) {
+      // if there is a related target, do nothing
+      if (event.relatedTarget !== null) return;
+      // if there is no related target re-focus the item
+      console.log('lost focus', event.target);
```
Comment (mportiz08) [inline src/components/Navigator/NavigatorCard.vue]: We should remove these debugging `console.log` statements before merging this in a PR.

File: src/components/Navigator/NavigatorCard.vue (hunk 1 above)
Comment (dobromir-hristov) [reply src/components/Navigator/NavigatorCard.vue]: I dont think so. We will probably iron out any bugs we notice, but from my testing it was alright.

**Review by hqhhuang — COMMENTED** [review]
1. Is it expected that the VO does not start at the item/page we are currently on?
For example, I was expecting VO to start on the sidenav at `Sloth.Color`, since this is the current component I'm on. However, it starts at the `init` under `Creating a Sloth`. If it starts on top of a page, I would have expected to start on  `Meet SlothCreator` and `Getting started with Sloths`. Is this a bug or pending future improvement?
Screen Shot 2022-02-18 at 4 34 47 PM

2. The VO is not working on the section titles like "Essentials" and "Creating a Sloth"

**Comment by marinaaisa** [comment]
> 1. Is it expected that the VO does not start at the item/page we are currently on?
>    For example, I was expecting VO to start on the sidenav at `Sloth.Color`, since this is the current component I'm on. However, it starts at the `init` under `Creating a Sloth`. If it starts on top of a page, I would have expected to start on  `Meet SlothCreator` and `Getting started with Sloths`. Is this a bug or pending future improvement?

I've not been able to reproduce this, sorry :(

> 2. The VO is not working on the section titles like "Essentials" and "Creating a Sloth"

That's because you are navigating through tabbing which only selects links. If you 
[Screen Recording 2022-02-21 at 16.34.33.mov.zip](https://github.com/dobromir-hristov/swift-docc-render/files/8110453/Screen.Recording.2022-02-21.at.16.34.33.mov.zip)
navigate using "Control + option + shift + arrow keys" you will be accessing also the text that are not links. I attach a video of it.

**Comment by hqhhuang** [comment]
For #1: 
[sidebar AX.mov.zip](https://github.com/dobromir-hristov/swift-docc-render/files/8122514/sidebar.AX.mov.zip)
In the screen recording, at first I was pressing the down arrow; And then after I refreshed the page, I was pressing tab. 

I tested with another page that is further down the tree (you would have to scroll up on the side bar to see the very top of the tree). In this scenario, the VO still starts at the very top of the tree, not just the top of what's visible on the page at the moment. I'm not sure if that's the intended behavior. I can see how starting from the top could make sense, because we are navigating through all the links on the page, and technically the first link is the beginning of the tree. But that doesn't seem like the best user experience. What do you think?

**Comment by marinaaisa** [comment]
> In the screen recording, at first I was pressing the down arrow; And then after I refreshed the page, I was pressing tab.

This behaviour when tabbing on Safari is expecting. Its documentation says "Tab also highlights buttons and other controls if “Highlight the next field or pop-up menu on a web page", which includes `button`s. While links are only highlighted when using "Option-Tab". You can read more here https://support.apple.com/en-gb/guide/safari/cpsh003/mac

> I tested with another page that is further down the tree (you would have to scroll up on the side bar to see the very top of the tree). In this scenario, the VO still starts at the very top of the tree, not just the top of what's visible on the page at the moment. I'm not sure if that's the intended behavior. I can see how starting from the top could make sense, because we are navigating through all the links on the page, and technically the first link is the beginning of the tree. But that doesn't seem like the best user experience. What do you think?

It really depends on how you navigate with VO but this issue should be address in this other radar instead: rdar://89153252

**Comment by mportiz08** [comment]
I am going to try to move this PR to be against the main apple/swift-docc-render repo since the base branch for this (`dhristov/add-sidenav`) has since been merged.

**Comment by mportiz08** [comment]
Looks like it won't let me switch the base branch outside of this repo. @dobromir-hristov we may need to re-open a new PR for this in the apple repo when you get back.

**Comment by hqhhuang** [comment]
Approving this PR, since as Marina mentioned, improvements will be addressed in rdar://89153252 instead.

> Looks like it won't let me switch the base branch outside of this repo. @dobromir-hristov we may need to re-open a new PR for this in the apple repo when you get back.

I think it's because we don't have write access to Dobri's repo. Marina should be able to rebase and merge tomorrow, since she was added as a collaborator on Dobri's fork.

**Comment by mportiz08** [comment]
He gave me write access before he left, so I can rebase and push commits to this PR—however, I don't think I can switch this to anything outside of the fork without opening a new PR. You're more than welcome to recreate the PR using the rebased branch though.

**Comment by dobromir-hristov** [comment]
We cannot change base branches across forks, that's a GitHub limitation.

Let me know, if I should just reopen a new one, or you do it instead :)

**Comment by marinaaisa** [comment]
Thanks everyone! I opened this new PR to be able to change the base branch :) https://github.com/apple/swift-docc-render/pull/55

**Pull request closed without merging**